salt/tests/unit/modules/test_metalk8s_kubernetes_utils.py

```diff
@@ -0,0 +1,190 @@
+        raises=False,
+        **kwargs
+    ):
+        """
+        Tests the return of `get_version_info` function
+        """
+        kubeconfig = MagicMock(return_value=tuple(kubeconfig))
+
+        apiclient = MagicMock(return_value=api_client)
+        api_instance = MagicMock()
+
+        if api_client is None or api_instance is None:
+            api_instance.return_value.get_code.side_effect = \
+                ApiException('Failed to get version info')
+        api_instance.return_value.get_code.return_value.to_dict.return_value  \
+            = result
+
+        with patch("metalk8s_kubernetes_utils.get_kubeconfig", kubeconfig), \
+                patch("kubernetes.config.new_client_from_config", apiclient), \
+                patch("kubernetes.client.VersionApi", api_instance):
+            if raises:
+                self.assertRaisesRegexp(
+                    CommandExecutionError,
+                    result,
+                    metalk8s_kubernetes_utils.get_version_info
+                )
+            else:
+                self.assertEqual(
+                    metalk8s_kubernetes_utils.get_version_info(**kwargs),
+                    result
+                )
+
+    def test_ping(self):
+        """
+        Tests the return of `ping` function
+        """
+        version_info = {"major": "1", "go_version": "go1.13.8"}
+        version_info_mock = MagicMock()
+
+        version_info_mock.return_value.get_version_info.return_value = \
+            version_info
+
+        with patch(
+            "metalk8s_kubernetes_utils.get_version_info", version_info_mock
+        ):
+
+            self.assertTrue(
+                metalk8s_kubernetes_utils.ping(),
+            )
+
+    @parameterized.expand([
+        param.explicit(kwargs=test_case)
+        for test_case in YAML_TESTS_CASES["read_and_render_yaml_file"]
+    ])
+    def test_read_and_render_yaml_file(
+        self, source, context, result, raises=False, template='jinja', **kwargs
+    ):
+        """
+        Tests the return of `read_and_render_yaml_file` function
+        """
+        cache_file_mock = MagicMock(return_value=source)
+        open_file_mock = mock_open(read_data=yaml.dump(source))
+        patch_dict = {
+            'cp.cache_file': cache_file_mock
+        }
+        with patch.dict(metalk8s_kubernetes_utils.__salt__, patch_dict), \
+                patch("salt.utils.files.fopen", open_file_mock):
+            if raises:
+                self.assertRaisesRegexp(
+                    CommandExecutionError,
+                    result,
+                    metalk8s_kubernetes_utils.read_and_render_yaml_file,
+                    source="my_source",
+                    template="my_template",
+                    **kwargs
+                )
+            else:
+                self.assertDictContainsSubset(
+                    result,
+                    metalk8s_kubernetes_utils.read_and_render_yaml_file(
+                        source=source,
```

Comment:
No ! 
`source` in this test is the content of the file where here you just want a path to the source file
So just put a random value 
```suggestion
                        source="my-source-file",
```